### PR TITLE
Use 32bit integer for loop indices in offload region.

### PR DESCRIPTION
**Pull request by ye-luo**
## Proposed changes
Close https://github.com/QMCPACK/qmcpack/issues/4774
Use int or uint32_t in offload loop indices.

## What type(s) of changes does this code introduce?
- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
epyc-server

## Checklist
- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

File: src/QMCWaveFunctions/BsplineFactory/SplineC2ROMPTarget.cpp [hunk 1]
```diff
@@ -866,7 +866,7 @@ void SplineC2ROMPTarget<ST>::mw_evaluateVGLandDetRatioGrads(const RefVectorWithL
             omptarget::min(last_cplx + omptarget::min(nComplexBands_local, last_cplx), requested_orb_size);
         ValueType ratio(0), grad_x(0), grad_y(0), grad_z(0);
         PRAGMA_OFFLOAD("omp parallel for reduction(+: ratio, grad_x, grad_y, grad_z)")
-        for (size_t j = first_real; j < last_real; j++)
+        for (int j = first_real; j < last_real; j++)
```
Comment (PDoakORNL) [inline src/QMCWaveFunctions/BsplineFactory/SplineC2ROMPTarget.cpp:869]: Why int for just a couple?

**Review by PDoakORNL — COMMENTED** [review]
Question about consistency

File: src/QMCWaveFunctions/BsplineFactory/SplineC2ROMPTarget.cpp (hunk 1 above)
Comment (ye-luo) [reply src/QMCWaveFunctions/BsplineFactory/SplineC2ROMPTarget.cpp:869]: A lot of places already has `int`. There is no need to change. I don't feel the need of 100% consistency. `int` for signed and `unit32_t` for unsigned. 64bit to 32bit is critical. signed vs unsigned is not.

**Review by PDoakORNL — APPROVED** [review]
This looks fine. I wish it were totally consistent.

I'd strongly urge that any integer variable including for indexes, that are going to be used to do more than simple indexing into an array i.e. have any signed math or interconversion with signed types just be signed from the start. The loss of one bit is worth avoiding these sort of bugs. 

An intersting detail we get here is if expression like `i * lda + j` are ever greater than the max uint32_t value they will wrap around since this is unsigned integer artithmetic.

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**